### PR TITLE
PF-1409: set workspace id and terra server id as default metadata when creating a notebook instance in WSM

### PR DESCRIPTION
**Pull request by yuhuyoyo**

### CODE REVIEW

**Review by zloery — APPROVED** [review]
LGTM

File: service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/CreateAiNotebookInstanceStepTest.java [hunk 1]
```diff
@@ -52,20 +55,20 @@ public void setFields() {
                 new ApiGcpAiNotebookInstanceContainerImage().repository("repository").tag("tag"));
 
     Instance instance =
-        CreateAiNotebookInstanceStep.setFields(creationParameters, "foo@bar.com", new Instance());
+        CreateAiNotebookInstanceStep.setFields(
+            creationParameters, "foo@bar.com", WORKSPACE_ID.toString(), SERVER_ID, new Instance());
     assertEquals("script.sh", instance.getPostStartupScript());
     assertEquals(true, instance.getInstallGpuDriver());
```
Comment (jaycarlton) [inline service/src/test/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/CreateAiNotebookInstanceStepTest.java:61]: nit: use `assertTrue`

**Review by jaycarlton — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/CreateAiNotebookInstanceStep.java [hunk 2]
```diff
@@ -185,6 +204,23 @@ static Instance setFields(
     return instance;
   }
 
+  private static void addDefaultMetadata(
+      Map<String, String> metadata, String workspaceId, String cliServer) {
+    if (metadata.put(WORKSPACE_ID_METADATA_KEY, workspaceId) != null) {
+      throw new BadRequestException(WORKSPACE_ID_METADATA_KEY + " metadata is reserved for Terra");
+    }
+    if (!StringUtils.isEmpty(cliServer)
```
[the remark this answers is not in the document]
Comment (yuhuyoyo) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/CreateAiNotebookInstanceStep.java]: it checks for null and empty

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/CreateAiNotebookInstanceStep.java [hunk 3]
```diff
@@ -185,6 +204,23 @@ static Instance setFields(
...
+        && metadata.put(SERVER_ID_METADATA_KEY, cliServer) != null) {
+      throw new BadRequestException(SERVER_ID_METADATA_KEY + " metadata is reserved for Terra");
```
[the remark this answers is not in the document]
Comment (yuhuyoyo) [reply service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/CreateAiNotebookInstanceStep.java]: conflict is more accurate.

**Pull request merged**